src/citrine/informatics/modules.py

```diff
@@ -0,0 +1,26 @@
+"""Tools for working with design spaces."""
+from typing import Any, List, Mapping, Type
+from uuid import UUID
+
+from citrine._serialization.polymorphic_serializable import PolymorphicSerializable
+from citrine._serialization.serializable import Serializable
+
+
+class Module(PolymorphicSerializable['Module']):
+    """A Citrine Module - an abstract type that returns the proper
+    subtype based on the 'type' value of the passed in dict.
+    """
+
+    _response_key = None
+
+    @classmethod
+    def get_type(cls, data) -> Type['Module']:
+        """Return the subtype."""
+        from citrine.informatics.design_spaces import DesignSpace
+        from citrine.informatics.processors import Processor
+        from citrine.informatics.predictors import Predictor
+        return {
+            'DESIGN_SPACE': DesignSpace,
+            'PROCESSOR': Processor,
+            'PREDICTOR': Predictor
+        }[data['module_type']].get_type(data)
```

Comment:
can this dictionary be a class-level variable instead of creating it then throwing it away on every call to this method?

i.e.

```
class Module(...):
  MODULE_TYPES = {'DESIGN_SPACE': DesignSpace, ... }

  @classmethod
  def get_type(cls, data):
    module = cls.MODULE_TYPES.get(data['module_type'])
    if module is not None:
      return module.get_type(data)
    raise Exception("{} is not a valid module type".format(data['module_type']))
```

Reply:
Yeah not a bad idea.

Reply:
Turns out I cannot do that with the dynamic import.